inst/include/Rcpp/exceptions.h

```diff
@@ -89,6 +89,48 @@ namespace Rcpp {
             file_io_error("file already exists", file) {}	// #nocov end
     };
 
+    #define RCPP_ADVANCED_EXCEPTION_CLASS(__CLASS__, __WHAT__)                                                                                                                                                     \
```

Comment:
Yes, something exactly like that is what I was thinking of :)

Indeed, I think we somewhat bluntly said "make tinyformat work in C++98 mode" a while back without consideration for C++11 mode, so it would make sense to guard the define there according to whether we're using C++98 or C++11.